### PR TITLE
New api

### PR DESCRIPTION
**Pull request by sjyk**
Added in the magical "Eval" operator.

### CODE REVIEW

**Comment by sirrice** [comment]
sweet!

File: src/main/scala/sampleclean/eval/Evaluator.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	  	  	  val totalTime = (System.currentTimeMillis-timeStart)/1000.0
+	  	  	  val accTuples:RDD[(Int,Int)] = result.map(compareValues(_,schema.length, schemaMap, unaryConstraints))
+	  	  	  val unaryResults = accTuples.reduce((x:(Int,Int),y:(Int,Int)) => (x._1+y._1,x._2+y._2))
+			  val binarySet = result.filter(x => binaryKeySet.contains(x(0).asInstanceOf[String]))
+			  val pairs = binarySet.cartesian(binarySet)
+			  val accBTuples:RDD[(Int,Int)] = pairs.map(x => compairPairs(x._1,x._2,schema.length, schemaMap, binaryConstraints))
+			  val binaryResults = accBTuples.reduce((x:(Int,Int),y:(Int,Int)) => (x._1+y._1,x._2+y._2))
+			  resultTable = ((unaryResults._2+0.0)/(unaryResults._1), 
+			  				 (binaryResults._2+0.0)/(binaryResults._1),
+			  				 totalTime) :: resultTable
```
Comment (thisisdhaas) [inline src/main/scala/sampleclean/eval/Evaluator.scala]: Can you add a docstring or something describing exactly what this function does? Looks like you're running through several algorithms on a sample and comparing accuracy of algorithm output on some passed in constraints? what does `resultTable` give? & etc...

Comment (sjyk) [inline src/main/scala/sampleclean/eval/Evaluator.scala]: Sure, will add documentation tonight. It was the most basic stats you could have. Accuracy on two types of metrics, either the attribute satisfies some value, or the attributes of two records are equal.

**Comment by thisisdhaas** [comment]
Awesome--that's very helpful.

**Pull request merged**